src/connector/src/sink/snowflake.rs

```diff
@@ -236,31 +227,87 @@ impl SnowflakeSinkWriter {
                 TimestamptzHandlingMode::UtcString,
                 TimeHandlingMode::String,
             ),
-            row_counter: 0,
-            payload: String::new(),
-            max_batch_row_num,
             // initial value of `epoch` will start from 0
             epoch: 0,
-        }
+            // will be initialized after the begin of each epoch
+            streaming_uploader: None,
+            file_suffix: None,
+            has_data: false,
+        })
     }
 
-    /// reset the `payload` and `row_counter`.
-    /// shall *only* be called after a successful sink to s3.
-    fn reset(&mut self) {
-        self.payload.clear();
-        self.row_counter = 0;
+    /// update the streaming uploader as well as the file suffix.
+    /// note: should *only* be called when a new epoch begins.
+    async fn update_streaming_uploader(&mut self) -> Result<()> {
+        self.file_suffix = Some(self.file_suffix());
+        let path =
+            generate_s3_file_name(self.s3_client.s3_path(), self.file_suffix.as_ref().unwrap());
```

Comment:
The `unwrap()` here is error-prone. See my previous comment.

Reply:
I not quite get it - why is it error-prone?

also for the previous comment, do you mean the in-accurate s3 path one - I don't see the relationship in between.

Reply:
> I don't get it - why is it error-prone?

Because it needs programmer to make sure `file_suffix` is not `None`, rather than being ensured by compiler. 

If you create a new `streaming_uploader` every time, the `file_suffix` will not need to be `Option`.